packages/js/src/classic-editor/helpers/dom.js

```diff
@@ -208,6 +209,25 @@ export const getPostMostUsedCategoryCheckboxes = () => {
 	return [ ...document.querySelectorAll( "#categorychecklist-pop input[type=checkbox]" ) ];
 };
 
+/**
+ * Maps over a list of checked checkboxes elements and returns the value and the text content.
+ *
+ * @param {Array} checkedCheckboxes The array of checked checkboxes to map.
+ *
+ * @returns {Object[]} An array containing objects with the checked checkboxes value and text content.
+ */
+const getValuesFromCheckboxes = ( checkedCheckboxes ) => {
+	return checkedCheckboxes.map( checkbox => (
+		{
+			id: checkbox.value,
+			name: [ ...checkbox.parentElement.childNodes ]
+				.filter( node => node.nodeType === Node.TEXT_NODE )
+				.map( node => node.textContent )[ 0 ]
+				?.trim(),
+		}
+	) );
+};
+
```

Comment:
For the reviewers:

I moved this code out of `getPostCategories` function. This piece of code is reused inside `getCustomTaxonomies` function, hence avoiding code duplication as well.